commands_buy.go

```diff
@@ -58,5 +59,49 @@ func (b buyCmd) ToAuditEntry() string {
 }
 
 func (b buyCmd) Execute() {
-	consoleLog.Warning("Not implemented: BUY")
+	consoleLog.Debug("Starting Buy")
+	if _, accountExists := accountStore[b.userID]; !accountExists {
+		consoleLog.Errorf("No account for %s exists", b.userID)
+		return
+	}
+
+	userAccount := accountStore[b.userID]
+
+	if userAccount.balance.ToFloat() < b.amount.ToFloat() {
+		consoleLog.Errorf("User %s does not have enough available funds for this purchase", b.userID)
+		return
+	}
+
+	qr := types.QuoteRequest{
+		Stock:      b.stock,
+		UserID:     b.userID,
+		AllowCache: true,
+		ID:         b.id,
+	}
+
+	var theQuote types.Quote
+	theQuote = getQuote(qr)
```

Comment:
What about the case where the returned quote will expire in 1s? IMO you should request an uncached quote before continuing with the transaction.